crates/uv-python/fetch-download-metadata.py

```diff
@@ -115,9 +120,13 @@ class PythonDownload:
     filename: str
     url: str
     sha256: str | None = None
+    variant: Variant | None = None
 
     def key(self) -> str:
-        return f"{self.implementation}-{self.version}-{self.triple.platform}-{self.triple.arch}-{self.triple.libc}"
+        if self.variant:
+            return f"{self.implementation}-{self.version}+{self.variant}-{self.triple.platform}-{self.triple.arch}-{self.triple.libc}"
```

Comment:
This `+variant` syntax is a little arbitrary — but I don't want to use `-` because I want the number of components to be constant. The alternative is using the short-form like `t` and `d` after the version e.g. `3.13.0t`, but I think this is clearer?

Reply:
This sort of steals the "local version" slot which could be problematic? Those aren't supported for Python versions though.